### ISSUE TITLE
[RESOURCE IDEA] Teaching accessibility to young coders

### ISSUE THREAD

**Issue by brianelton** [w3c-cg/aar#5]
In the educational space, a lot of emphasis has been put recently on how to teach accessibility at the collegiate or university level, often in computer science courses. I am proposing that we should be looking at how to incorporate accessibility into the first instances that students learn about coding, often in middle school.
The challenge with teaching accessibility to young coders is that most accessibility topics will not resonate, either from a lack of coding knowledge or lack of understanding of the more complex user interactions that they are not exposed to yet.

I am proposing the creation of a resource that teachers and other educators of kids beginning to learn coding can use to:

1. properly frame the subject into understandable topics and principles;
2. focus on the aspects of accessibility that would apply to the type of coding that they are learning;
3. motivate the teachers and students to include the basic principles of accessibility from the very start.

I have done some research on this topic that I am intending to use for a conference presentation, but would like help developing the resource that such talks can refer to and direct people to.

Adding link to Google doc where ideas are being fleshed out: https://docs.google.com/document/d/1aFU-RyLrIdGZGuZbDWn_IoT5Cal8RBQhF9uyJezS2Cc/edit?usp=sharing

**Comment by brianelton** [comment]
@jade-mc I thought of you specifically, considering your PHD topic of teaching accessibility in higher education. Perhaps there is some continuity that can be built in.

**Comment by jade-mc** [comment]
Hi Brian, 

I think there's actually a fair bit of research into how (successfully) accessibility has been embedded into computer science, if you have a look on the [ACM database](https://www.acm.org/) then a fair amount should come up. 
My research is more focused on how we can teach staff in higher ed (especially those outside of the teaching and learning space) about accessibility, what they need to know, so what constitutes a core curriculum, different models of learning (pedagogy/andragogy) and how communities of practice can support learning. I'll be narrowing that down next year. 

Jade

**Comment by brianelton** [comment]
> I think there's actually a fair bit of research into how (successfully) accessibility has been embedded into computer science

Is this at the post-secondary school level? In all of my searching (with some help with Michelle Williams) I have not been able to find any research on teaching accessibility to 10-12 year olds.

**Comment by jade-mc** [comment]
Well in higher education, certainly. There is a project looking into kids here in the UK, I can't remember the name of the researcher, I'll try!

**Comment by jade-mc** [comment]
This is what I was thinking about! https://www.learntoenable.co.uk/ it isn't quite coding, but it's kids.

**Comment by lbeaze** [comment]
Scope question: Is the goal to create materials to be used by teachers in the class or materials to help teachers adapt their curriculum?

**Comment by brianelton** [comment]
> Scope question: Is the goal to create materials to be used by teachers in the class or materials to help teachers adapt their curriculum?

Yes, I think this should be included in the scope. One of the major hurdles is that the middle school teachers don't have the knowledge themselves, so having materials for them would be very helpful.

**Comment by lsr-explore** [comment]
@brianelton  - I am interested in helping.   What age group are you targeting?

I'd like to see included some discussion of designing software products for people with disabilities.   It has been exciting to see the trend from "let's fix accessibility by teaching the engineers how to code for accessibility" to "let's include accessibility when we are thinking about the product definition and user experience.   I think we could include resources for helping future leaders, product managers, user research and user experience designers to take an accessibility first approach.

That said, I'm happy to support whatever areas the group feels are important.

## Ideas
1) I think that  scratch is a popular platform for young coders - I haven't used it - found this conference talk - haven't watched the talk yet
Scratch Conference 2023: Developing Accessibility Tools
https://www.youtube.com/watch?v=m-4YKkR9wvE

This article from the scratch team about making colors changes to improve the accessibility of the tools
https://en.scratch-wiki.info/wiki/Accessibility_Color_Changes

---------
2) Creating lessons showing kids how to evaluate the accessibility of a site - using keyboard and dev tools.   Not sure if this would be too advanced.
----------
3) I have found when trying to demonstrate accessibility to colleagues, that learning through example is helpful.
-----------
4) I don't know if kids learn about writing tests, but it is an important aspect to coding.
----------
5) I have just started trying out cody - ai extension in visual studio code, and found that at times it includes accessibility when it generates code, but including accessibility in the prompts is also useful.   I wonder if the future of coding for our youth will be mostly writing prompts :-)

**Comment by brianelton** [comment]
Cool example of incorporating a11y into a playful coding experience from Apple - https://developer.apple.com/videos/play/wwdc2020/10681/

**Comment by ktang-ux** [comment]
In my recent work event, we had a speaker comes in to demonstrate what they would hear from a screen reader and that sparks a lot of conversation among the company. I think something along that line would make younger kids be more empathetic which leads to their willingness to code with accessibility in mind. My coworker and I had been brainstorming on ideas for activities like wear oven mitts or blindfold to complete a task. Maybe we can come up with a list of them for teachers to use or setup up a table with a few of these exercise while someone's out doing a talk.

**Comment by HelenL2E** [comment]
> This is what I was thinking about! https://www.learntoenable.co.uk/ it isn't quite coding, but it's kids.

Hi there, I was steered to this conversation from a post in LinkedIn. 

I am the one behind LearnToEnable which is my PHD work. In my research, I'm not looking at coding I'm looking more at the everyday principles of digital accessibility, e.g. building on the model and work I did in 2019 called SCULPT for Accessibility, only adding extras such as captions, transcripts, fonts and reading order to make it more applicable to broader digital content. 
https://abilitynet.org.uk/news-blogs/everyone-can-sculpt-accessibility

My PHD research is specifically aimed at digital accessibility awareness at upper Key Stage 2 (UKS2), which is Year 5&6, (more precisely aged 9-10). My PHD is to teach the basic everyday principles such as headings, links, alt text, captions, colour contrast, use of colour etc. 

At present we have a young generation growing up with little or no knowledge of these basics, never mind the coding side of it. I didn't want these skills to sit specifically as a niche set of coding skills, but everyday skills kids can easily apply throughout everyday life e.g. if they make a documents, post on social media as they get older etc. I have positioned it as skills for both computing and citizenship, e.g. responsible and inclusive citizens of our digital world, and with basics that can be easily applied by anyone.

I'm working collaboratively with UKS2 teachers to build the curriculum materials, so as part of the research they will get trained, then adapt the training to create lessons suitable for UKS2 learners. I'll be looking at both the creation and development of the materials, but also the development of teachers pedagogical content knowledge as they go through cycles of collaborative lesson study.

My overall aim is to create resources for teachers by teachers so the basics of digital accessibility awareness can be embedded or included as a norm.

Aside form my PHD I've also been doing some work in FE colleges, as well as tested some of the content in a high school.

I'd be really interested in your PHD work, I'm on LinkedIn as Helen Wilson if you would like to connect, and to save any confusion I have used my pseudonym of Georgie K Haynes on my Learn to Enable website.

**Comment by brianelton** [comment]
Hi @HelenL2E ! Thank you for reaching out. I think the work that you have done with Learn to Enable is fantastic! And it's the only other resource I have been able to find that focuses on teaching accessibility concepts to children, so it has also been inspirational.
While there is certainly some overlap between your work and what I am proposing here, my primary target is teaching accessibility in a coding environment. I think it needs to be done in a slightly different way than the way you are approaching teaching accessibility principles (this different way is part of what I am exploring). I expect and hope our work will complement each other well.    
Just as a note, this work is not being done as part of a PhD or other academic research - I am an accessibility professional who teaches other professionals about digital accessibility. I found that there is a large gap in early education when it comes to accessibility and wanted to explore it further. This W3C community group that the work is being done within comprises mostly of other (as far as I know) non-academic accessibility professionals (@jade-mc being a noted exception to that). 
I will connect with you on LinkedIn and look forward to discussing this further!

**Comment by Different2day** [comment]
Hi Brian, 

Anya here.

I don't have feedback on the scope document but am wondering whether the work I'm doing on the accessibility knowledge graph could be of any use.

It's about looking at all the parts that make up accessibility. Signals, signs, sign systems, concepts that are being communicated, system features, design, code, user activity, communication goals, perception, cognitive abilities, motor skills, familiarity, context - to show how accessibility sits at the intersection of all that. The aim is to have it all aligned with WCAG and supplemental guidance for cognitive accessibility.

That's nothing new, I'm sure, it's just a matter of getting it all into a graph database. And once it's in there, it could perhaps help structure communication and teaching about it all. Not sure if this is of any use to you but I thought I'd mention it.

**Comment by brianelton** [comment]
I have started to add thoughts and some structure in a Google doc - https://docs.google.com/document/d/1aFU-RyLrIdGZGuZbDWn_IoT5Cal8RBQhF9uyJezS2Cc/edit?usp=sharing
If you would like to contribute, follow the link and activate the request access link.

**Comment by brianelton** [comment]
Closing issue. The AAR CG will be starting a new direction soon.
